sklearn/ensemble/_hist_gradient_boosting/grower.py

```diff
@@ -554,6 +613,45 @@ def split_next(self):
 
         return left_child_node, right_child_node
 
+    def _compute_interactions(self, node):
+        r"""Compute features allowed by interactions to be inherited by child nodes.
+
+        Example: Assume constraints [{0, 1}, {1, 2}].
+           1      <- Both constraint groups could be applied from now on
+          / \
+         1   2    <- Left split still fulfills both constraint groups.
+        / \ / \      Right split at feature 2 has only group {1, 2} from now on.
+
+        Parameters:
+        ----------
+        node : TreeNode
+            A node that might have children. Based on its feature_idx, the interaction
+            constraints for possible child nodes are computed.
+
+        Returns
+        -------
+        allowed_features : None or ndarray, dtype=int
+            Indices of features allowed to split for children.
+        interaction_cst_indices : None or list of ints
```

Comment:
```suggestion
        interaction_cst_indices : list of ints
```

Reply:
Why remove `None` as valid option?

Reply:
In the current implementation, the return is always a list and never `None`.

Reply:
🙈 That happens if you look into your own code 1 year later 🙈